packages/core/src/amazonqGumby/telemetry/codeTransformTelemetryState.ts

```diff
@@ -34,6 +43,8 @@ class CodeTransformerTelemetryState {
     public getSessionId = () => this.mainState.sessionId
     public getStartTime = () => this.mainState.sessionStartTime
     public getResultStatus = () => this.mainState.resultStatus
+    public getCodeTransformMetaData = () => this.mainState.codeTransformMetadata
+    public getCodeTransformMetaDataString = () => JSON.stringify(this.mainState.codeTransformMetadata)
```

Comment:
how big is this payload, potentially? and does it potentially contain PII?

Reply:
The `codeTransformMetadata` is the start of the new metrics to switch away from the current telemetry implementation. We, we will add a safeguard for truncating this for the max column length in the backend. There is a whole internal proposal from our BI team that has been passed around a couple months back. We are testing it out for this project.

There is not any PII data, we are not logging any user. Only success and generic failure detail messages.